parsec/backend/postgresql/factory.py

```diff
@@ -51,23 +51,29 @@ async def _send_event(
     block = PGBlockComponent(dbh, blockstore, vlob)
     events = EventsComponent(realm, send_event=_send_event)
 
+    components = {
+        "events": events,
+        "webhooks": webhooks,
+        "http": http,
+        "organization": organization,
+        "user": user,
+        "invite": invite,
+        "message": message,
+        "realm": realm,
+        "vlob": vlob,
+        "ping": ping,
+        "block": block,
+        "blockstore": blockstore,
+    }
+    for component in components.values():
+        method = getattr(component, "register_components", None)
```

Comment:
So `register_components` so different components can know about one another? And so far it's only used for `http` to know about `organization`?

Reply:
yes, because so far the postgresql components rely on SQL requests to access all the data (i.e. unlike the memory implementation, no component owned the underlying data).

However the administration api implemented in the `http` component is very closely related to the `organization` component.
This is because `http` component is related to a way to access data (i.e. through HTTP, hence the REST API being put there), where all the others components provide way to manipulate data and way to access data through the websocket API.

In theory we should have instead:
- a class exposing way to manipulate data
- a class exposing websocket API (maybe even a class per authentication type !)
- a class exposing HTTP API
- `http` module would only contain a class exposing the static HTTP routes
- `organization` module would contain all the three class types

However this seems too cumbersome to implement for the moment given how small the REST API currently is, so using the component registration system seems a much simpler solution ;-)